src/test/java/org/opensearch/ad/transport/ValidateAnomalyDetectorTransportActionTests.java

```diff
@@ -435,4 +423,47 @@ public void testValidateAnomalyDetectorWithDetectorNameTooLong() throws IOExcept
         assertEquals(ValidationAspect.DETECTOR, response.getIssue().getAspect());
         assertTrue(response.getIssue().getMessage().contains("Name should be shortened. The maximum limit is"));
     }
+
+    @Test
+    public void testValidateAnomalyDetectorWithNonExistentTimefield() throws IOException {
```

Comment:
Are you working on any tests to run the validation API with sparse data to test out all of the different scenarios? I think it's ok to have as a TODO, but maybe open an issue to track and add for following release.

Reply:
Yes I am currently continuing to do it in terms of integration tests like what I added here: `src/test/java/org/opensearch/ad/e2e/DetectionResultEvalutationIT.java`. Will cut an issue and work on adding more in separate PR.